src/silentpayment.d.ts

```diff
@@ -0,0 +1,53 @@
+/// <reference types="node" />
+import { bip341 } from '.';
+export declare type Target = {
+    silentPaymentAddress: string;
+    value: number;
+    asset: string;
+};
+export declare type Output = {
+    scriptPubKey: string;
+    value: number;
+    asset: string;
+};
+export interface TinySecp256k1Interface extends bip341.BIP341Secp256k1Interface {
+    privateMultiply: (key: Uint8Array, tweak: Uint8Array) => Uint8Array;
+    pointMultiply: (point: Uint8Array, tweak: Uint8Array) => Uint8Array | null;
+    pointAdd: (point1: Uint8Array, point2: Uint8Array) => Uint8Array | null;
+    pointFromScalar: (key: Uint8Array) => Uint8Array | null;
+    privateAdd: (key: Uint8Array, tweak: Uint8Array) => Uint8Array | null;
+    privateNegate: (key: Uint8Array) => Uint8Array;
+    ecdh: (pubkey: Uint8Array, privkey: Uint8Array) => Uint8Array;
+}
+export declare class SilentPaymentAddress {
+    readonly spendPublicKey: Buffer;
+    readonly scanPublicKey: Buffer;
+    constructor(spendPublicKey: Buffer, scanPublicKey: Buffer);
+    static decode(str: string): SilentPaymentAddress;
+    encode(): string;
+}
+export declare class SilentPayment {
+    private ecc;
+    constructor(ecc: TinySecp256k1Interface);
+    /**
+     * create the transaction outputs sending outpoints identified by *outpointHash* to the *targets*
+     * @param inputsOutpointsHash hash of the input outpoints sent to the targets
+     * @param sumInputsPrivKeys sum of input private keys
+     * @param targets silent payment addresses receiving value/asset pair
+     * @returns a list of "silent-payment" taproot outputs
+     */
+    pay(inputsOutpointsHash: Buffer, sumInputsPrivKeys: Buffer, targets: Target[]): Output[];
```

Comment:
this function sig is too complicated.. not user friendly at all. What if we expect here a list of inputs (outpoints) and we hash them internally? Same for the prvkey sum, let's expect a list of prv keys and we make the sum.

This way we might not need to export `outpointHash` (totally unexpected to see an exported function like that in this module)?

Reply:
> What if we expect here a list of inputs (outpoints) and we hash them internally? Same for the prvkey sum, let's expect a list of prv keys and we make the sum.

isn't better to do the "signing operation" outside of the function ? we moved signing process outside of Pset so the idea was to do the same here with "private keys aggregation". I would keep this way for private keys.

about outpointsHashes it's the same concern: passing the hash directly let's to do it outside if necessary.

> this function sig is too complicated

I would enrich other liquidjs-lib classes using this comlicated one. Some ideas:
- `Updater.addSilentPayments(inputsIndexes: number[], targets: Target[]}`
- `payments.p2sp({ scanKey, spendKey })`
- ??

what do u think ?

Reply:
> I would enrich other liquidjs-lib classes using this one. Some ideas:
>
> * Updater.addSilentPayments(inputsIndexes: number[], targets: Target[]}
> * payments.p2sp({ scanKey, spendKey })
> * ??
>
> what do u think ?

We won't add an ad hoc method to psetv2 for adding silent payments just like we didn't for taproot or any other kind of output.
On the contrary, we can expose some well-thought silent payment APIs , just like done for taproot or other script types (payment, address), that make managing a transaction (psetv2) easy.